src/topology/running.rs

```diff
@@ -347,15 +361,18 @@ impl RunningTopology {
         let conflicting_sinks = conflicts
             .into_iter()
             .filter(|&(existing_sink, _)| existing_sink)
-            .map(|(_, name)| name.clone());
+            .map(|(_, id)| id.clone());
 
         // Buffer reuse
         // We can reuse buffers whose configuration wasn't changed.
         let reuse_buffers = diff
             .sinks
             .to_change
             .iter()
-            .filter(|&name| self.config.sinks[name].buffer == new_config.sinks[name].buffer)
+            .filter(|&id| {
+                self.config.sinks.get(id).map(|item| &item.inner.buffer)
+                    == new_config.sinks.get(id).map(|item| &item.inner.buffer)
+            })
```

Comment:
Here as well, I think. Previously we'd blow up if something is missing that we expect to be present, and now it'll pass silently. Likely not a huge deal, just curious what the motivation is behind changing the behavior.

Reply:
same problem, indexing with a struct 🤔